mutation_partition.cc

```diff
@@ -146,7 +146,14 @@ mutation_partition::mutation_partition(const schema& s, const mutation_partition
         , _static_row(s, column_kind::static_column, x._static_row)
         , _static_row_continuous(x._static_row_continuous)
         , _rows()
-        , _row_tombstones(x._row_tombstones) {
+        , _row_tombstones(x._row_tombstones)
+#ifdef SEASTAR_DEBUG
+        , _schema_version(s.version())
+#endif
+{
+#ifdef SEASTAR_DEBUG
+    assert(x._schema_version == _schema_version);
+#endif
```

Comment:
how about defining SEASTAR_DEBUG_ASSERT to improve readability?

Reply:
It is not really related to Seastar. Scylla should define its own fine-grained debug macros.

For now it is okay to hitch a ride in SEASTAR_DEBUG.